### PR TITLE
#1197 Get or Set settings data when using ketcher-react 

### PR DESCRIPTION
**Pull request by ElenaOdnoshivkina**

### CODE REVIEW

**Review by AndreiMazol — CHANGES_REQUESTED** [review]
need to  discuss

File: packages/ketcher-core/src/domain/services/storage/storage.types.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ***************************************************************************/
+
+type RecordOrArrayOfRecords =
+  | Record<string, unknown>
+  | Record<string, unknown>[]
+export interface StorageProvider {
+  set: (data: RecordOrArrayOfRecords, key: string) => Promise<any>
```
Comment (AndreiMazol) [inline packages/ketcher-core/src/domain/services/storage/storage.types.ts:21]: set/get should be generics

File: packages/ketcher-react/src/script/builders/ketcher/KetcherBuilder.ts [hunk 2]
```diff
@@ -32,12 +33,14 @@ class KetcherBuilder {
...
   private serviceMode: ServiceMode | null
   private formatterFactory: FormatterFactory | null
+  private storage: StorageProvider | undefined
 
   constructor() {
     this.structService = null
     this.editor = null
     this.serviceMode = null
     this.formatterFactory = null
+    this.storage = undefined
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/builders/ketcher/KetcherBuilder.ts:43]: ?

File: packages/ketcher-react/src/script/builders/ketcher/KetcherBuilder.ts [hunk 3]
```diff
@@ -54,11 +57,20 @@ class KetcherBuilder {
...
+    return this.storage
+  }
+
   async appendUiAsync(
     element: HTMLDivElement | null,
     staticResourcesUrl: string,
     errorHandler: (message: string) => void,
-    buttons?: ButtonsConfig
+    buttons?: ButtonsConfig,
+    storage?: StorageProvider
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/builders/ketcher/KetcherBuilder.ts:73]: use correct variable names

**Review by AndreiMazol — CHANGES_REQUESTED** [review]
need to discuss

**Comment by Mariia-Katkova** [comment]
In 2.5 you can save settings already

**Pull request closed without merging**